site/app/controllers/MiscController.php

```diff
@@ -115,11 +115,6 @@ public function displayFile($dir, $path, $gradeable_id = null, $user_id = null,
                     return false;
                 }
             }
-
-            if (!$this->core->getUser()->accessGrading() && !CourseMaterial::isSectionAllowed($this->core, $path, $this->core->getUser())) {
```

Comment:
Why is check removed?

Reply:
This check is meant for the courseMaterial files but currently it is being used for submission file also, which break the submitty. Also this check is already in place for the courseMaterial file (see above `if` conditions Line no. 107 and 113).

Reply:
> I still don't understand why the check for section-specific course materials is being removed.
> I'm not sure what you're talking about with the if condition on 107, 113, thats the condition you're removing in this PR.

@shailpatels `isSectionAllowed` check is only meant for the courseMaterial files and currently, we are testing submission files also against this check which is not correct (please correct me if I'm wrong?)  
Also, this check is causing submitty to crash when you try to view the submission files (i.e using `display_file` function). 
On line 107 and 113 I was quoting that this condition( `isSectionAllowed`) is already present for the CoursMaterial files i.e ` if ($dir == 'course_materials') `. In line 113 you'll find that this is the exact same code (or to say,  `isSectionAllowed` condition) which is good and it should be as it is. So, that's the reason behind removing the check (buggy).

Reply:
Testing course materials, removing this check does not cause any new problems.